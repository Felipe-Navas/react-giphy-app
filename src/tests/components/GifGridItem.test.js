import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Testing the GiftGridItem component", () => {

  const title = 'Titulo1';
  const url = 'https://localhost/something.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  
  test("should show the component GifGridItem", () => {

    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with the parameter title', () => { 
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
   });

  test('should have an image with the alt and src', () => {
    const img = wrapper.find('img');
    
    //console.log(img.props().alt);
    //console.log(img.prop('alt'));
    
    expect(img.prop('alt')).toBe(title);
    expect(img.prop('src')).toBe(url);

    });

    test("should have the classname animate__fadeIn in the div", () => {
      const div = wrapper.find("div");

      //console.log(div.props().className);

      const className = div.props().className;;
      expect(className.includes("animate__fadeIn")).toBe(true);
    });
});
