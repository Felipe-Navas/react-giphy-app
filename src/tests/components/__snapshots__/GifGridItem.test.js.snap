// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the GiftGridItem component should show the component GifGridItem 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo1"
    src="https://localhost/something.jpg"
  />
  <p>
     
    Titulo1
     
  </p>
</div>
`;
